client/state/utils.js

```diff
@@ -375,3 +375,35 @@ export function combineReducersWithPersistence( reducers ) {
 	combined.hasSchema = validatedReducersHasSchema;
 	return combined;
 }
+
+/**
+ * Wraps a reducer such that it won't persist
+ * any state to the browser's local cache
+ *
+ * @example revent a simple reducer from persisting
+ * const age = ( state = 0, { type } ) =>
+ *   GROW === type
+ *     ? state + 1
+ *     : state
+ *
+ * export default combineReducers( {
+ *   age: withoutPersistence( age )
+ * } )
+ *
+ * @example preventing a large reducer from persisting
+ * const posts = withoutPersistence( keyedReducer( 'postId', post ) )
```

Comment:
If we could avoid the caller from having to think about persistence if they don't want it, it would be ideal. Basically I'm thinking this would only be called by other util methods, especially if #13542 lands. 

I'm guessing you'll update `keyedReducer` to handle this already in #14007